src/tests/system/tests/test_cache.py

```diff
@@ -146,66 +147,159 @@ def test_cache__writes_to_both_database_files_when_using_fully_qualified_names(
+    """
+    group1 = provider.group("group1").add()
+    group2 = provider.group("group2").add()
+    client.sssd.start()
+
+    assert client.tools.getent.group("group1"), "Group group1 not found!"
+    assert client.tools.getent.group("group2"), "Group group2 not found!"
+
+    group1.add_member(provider.user("user1").add())
+    group2.delete()
+
+    client.sssctl.cache_expire(everything=True)
+
+    assert client.tools.getent.group("group1"), "Group group1 not found!"
+    assert "user1" in str(client.tools.getent.group("group1")), "Group group1 is missing a user!"
+    assert not client.tools.getent.group("group2"), "Group group2 found!"
+
+
+@pytest.mark.integration
+@pytest.mark.importance("low")
+@pytest.mark.parametrize("obj", ["user", "group", "netgroup"])
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_cache__invalidate_entries_in_domain_and_timestamps_caches(
+    client: Client, provider: GenericProvider, obj: str
+):
+    """
+    :title: Invalidates object entries in the domain and timestamps caches using dataExpireTimestamp attribute
+    :setup:
+        1. Create user, group and netgroup
+        2. Add user as a member to group and netgroup
+        3. Start SSSD and lookup user, group and netgroup
+    :steps:
+        1. Search for object attribute dataExpireTimestamp in domain and timestamps ldb caches
+        2. Clear the SSSD cache for the object
+        3. Compare the dateExpireTimestamp value against previous search
+    :expectedresults:
+        1. Attribute is found
+        2. Cache is cleared for the object
+        3. Attribute is found and the value has changed
+    :customerscenario: False
+    """
+    # Unable to parametrize provider method, so all object types are created
+    # The object names are specific matching the suffix, cannot be renamed without changing the search query.
+    user = provider.user("user").add()
+    provider.group("group").add().add_member(user)
+    provider.netgroup("netgroup").add().add_member(user=user)
+    client.sssd.start()
+
+    assert client.tools.getent.passwd("user")
+    assert client.tools.getent.group("group")
+    assert client.tools.getent.netgroup("netgroup")
+
+    cache = f"/var/lib/sss/db/cache_{client.sssd.default_domain}.ldb"
+    timestamps = f"/var/lib/sss/db/timestamps_{client.sssd.default_domain}.ldb"
+    suffix = f"cn={client.sssd.default_domain},cn=sysdb"
+
+    # Users and groups objects are cached as the fully qualified name unlike netgroups
+    if obj != "netgroup":
+        obj = f"{obj}@{client.sssd.default_domain}"
+
+    for dbs in [cache, timestamps]:
+        result = client.ldb.search(dbs, suffix, filter=f"name={obj}")
+        assert result is not None, f"{obj} object not found!"
+        for k, v in result.items():
+            if isinstance(k, dict):
+                expire_time = result.get(str(k)).get("dataExpireTimestamp")
+                client.host.conn.exec(["sss_cache", f"-{obj[0]}", obj])
```

Comment:
First 'sss_cache' (with 'dbs == cache') invalidates entry in both databases.
So when test gets to 'dbs == timestamps', entry should be already invalidated and 'expire_time' should be already reset.
Not sure how it works..?

Reply:
Yes, it's supposed to. The expire_time variable contains the value before clearing the cache and it's compared against the new dataExpireTimestamp.

Reply:
On a 2nd run - when 'dbs == timestamps' - 'expire_timestamp' is set to a value from a cache that was already cleared on a previous run, when 'dbs == cache'.

Reply:
@danlavu, ping

Reply:
Sorry, this was put on the back burner. Yes, I see the issue now.  I also added another assertion to ensure that the value is not empty, it was skipping the assertion and passing test because no assertion was made.